src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,494 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import {
+  getRandomValue,
+  useIntersection,
+  useMergedRefs,
+  useTheme,
+} from '../utils';
+import { IconButton } from '../Buttons';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/carousel.css';
+
+export type CarouselProps = {
+  /**
+   * Index of the currently shown slide.
+   * Can be used to set the default index or control the active slide programmatically.
+   * @default 0
+   */
+  activeSlideIndex?: number;
+} & React.ComponentPropsWithoutRef<'div'>;
+
+export const CarouselContext = React.createContext<
+  | {
+      /** Stateful index of currently active slide. */
+      currentIndex: number;
+      /** State updater for currentIndex. */
+      setCurrentIndex: (index: number | ((old: number) => void)) => void;
+      /** Number of slides in the carousel. Gets set in CarouselSlider for reading in CarouselDots.  */
+      slideCount: number;
+      /** State updater for slideCount. */
+      setSlideCount: (length: number | ((old: number) => void)) => void;
+      /** Stateful value of the left/right arrow keys currently pressed. */
+      keysPressed: Record<string, boolean>;
+      /** Prefix used for setting id for internal carousel components. */
+      idPrefix: string;
+      /** Ref object used for preventing intersection observer from breaking manual slide updates. */
+      isManuallyUpdating: React.MutableRefObject<boolean>;
```

Comment:
I still don't understand this prop.

Reply:
How about now?
![image](https://user-images.githubusercontent.com/9084735/158691307-8443247d-00c5-4089-ac94-61cba6190d36.png)